### ISSUE TITLE
Component remounts when `app.set` is called

### ISSUE THREAD

**Issue by daryl** [dekujs/deku#172]
I'm not sure if this is an issue or desired behaviour, so apologies if it's the latter.

Here's an example of what's going on, I took out what I considered unnecessary, hopefully you can get an understanding of the pattern:

```js
function plugin(app) {
  bus.on('foo', () => {
    app.set('foo', 'bar');
  });
}

const Page = {
  afterMount() {
    console.log('Mounted'); // Will log more than once

    setTimeout(() => {
      bus.emit('foo');
    }, 500);
  },

  render() {
    return (
      <div>Page</div>
    );
  }
};

const App = {
  propTypes: {
    foo: {source: 'foo'}
  },

  render({props}) {
    let {foo} = props;

    if(foo) {
      foo = <div>{foo}</div>;
    }

    return (
      <div>
        {foo}
        <Page/>
      </div>
    );
  }
};
```

It seems that when the `App` structure changes it will remount `Page`.

**Comment by anthonyshort** [comment]
I think this is the same thing as #163. It's mounting again because it's moving in the DOM from position 0 to 1 and back again, and it doesn't know that it's the same one. You could either wrap it in a div that doesn't move or you could add a `key` (but all siblings need keys)

```js
<div>
  <div>{foo}</div>
  <Page/>
</div>
```

I think this case would be solved if we can handle null/undefined elements as just empty elements, like a `<noscript />` element.

**Comment by anthonyshort** [comment]
This is fixed with `0.5.0` when you render `false` or `null` when `foo` doesn't exist.
